snuba/migrations/snuba_migrations/transactions/0002_transactions_onpremise_compatibility.py

```diff
@@ -0,0 +1,179 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import (
+    Column,
+    DateTime,
+    LowCardinality,
+    Materialized,
+    Nullable,
+    String,
+    UInt,
+    WithDefault,
+)
+from snuba.clusters.cluster import ClickhouseClientSettings, get_cluster
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+
+
+UNKNOWN_SPAN_STATUS = 2
+
+
+def recreate_table() -> None:
+    """
+    The sample by clause for the transactions table was added in April 2020.
+    If the user has a table without the sample by clause, we need to add it by
```

Comment:
Is this done in order to replace the DROP TABLE from the onpremise install in new versions? Since now whoever has the old table gets it rebuilt during innstall and if the table is full they cannot go ahead.

Reply:
Yes, I think we shouldn't keep that workaround in onpremise forever, this is supposed to be a replacement.